src/particles/profiles/GetInitialDensity.cpp

```diff
@@ -37,4 +41,9 @@ GetInitialDensity::GetInitialDensity (const std::string& name)
             }
         }
     }
+    if (m_profile == BeamProfileType::Parsed) {
+        std::string density_func_str = "0.";
+        getWithParser(pp, "density(x,y,z)", density_func_str);
+        m_density_func = makeFunctionWithParser<3>(density_func_str, m_parser, {"x", "y", "z"});
+    }
```

Comment:
Please combine the two if-else blocks to clean this up.
https://github.com/Hi-PACE/hipace/pull/912/files#diff-5dee9dcb76a6489818f7175cfcf68ddebd0c5e74b2c131315cc006c0900bb420R17-R48